### ISSUE TITLE
Catch parameter count mismatch in `with-redefs`

### ISSUE THREAD

**Issue by mrkam2** [clj-kondo/clj-kondo#2126]
To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

When I'm changing function signatures, in particular parameter counts, linter currently doesn't catch outdated `with-redefs` for such functions.

Here is an example:
```
(defn x [a b] (+ a b))

(defn y [] (x 10 20))

(with-redefs [x (fn [a] a)] (y))
```

Would be great if it was flagging at least simpler cases like the one above.

It is of course less trivial for vararg functions or functions with multiple arities. But I believe we should at least match one of the existing arities in `with-redefs`.

**Comment by borkdude** [comment]
Why would you rebind a var to a function of a different arity? Doesn't make sense to me, please explain.

**Comment by mrkam2** [comment]
> Why would you rebind a var to a function of a different arity? Doesn't make sense to me, please explain.

That's exactly is the reason for a warning. `with-redefs` is often used in tests to mock the existing functionality. And when that changes (for example, I add or remove a parameter from a function), sometimes these mock functions are forgotten to be updated. Linter can potentially catch that before the tests are executed.

**Comment by borkdude** [comment]
So it should warn when you override with a function of different arity? Makes sense

**Comment by mrkam2** [comment]
Would still like to have this fixed.
